lib/beaker/host.rb

```diff
@@ -344,7 +344,8 @@ def mkdir_p dir
       if self.is_cygwin?
         cmd = "mkdir -p #{dir}"
       else
-        cmd = "if not exist #{dir.gsub!('/','\\')} (md #{dir.gsub!('/','\\')})"
+        windows_dirstring = dir.gsub!('/','\\')
```

Comment:
Also `gsub!` modifies the `dir` parameter passed into this method, and would be surprising to the caller.